graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ConfigGenerator.java

```diff
@@ -0,0 +1,304 @@
+                            clientConfigTypesList.contains(((TypeDefinitionNode) node).typeName().text())) {
+                        if (((TypeDefinitionNode) node).typeName().text().equals("ConnectionConfig")) {
+                            node = constructConnectionConfig(node, authConfig);
+                        }
+                        memberDeclarationNodes.add(node);
+                    }
+                } else if (authConfig.isClientConfig()) {
+                    if (commonTypesList.contains(((TypeDefinitionNode) node).typeName().text()) ||
+                            clientConfigTypesList.contains(((TypeDefinitionNode) node).typeName().text())) {
+                        if (((TypeDefinitionNode) node).typeName().text().equals("ConnectionConfig")) {
+                            node = constructConnectionConfig(node, authConfig);
+                        }
+                        memberDeclarationNodes.add(node);
+                    }
+                } else if (commonTypesList.contains(((TypeDefinitionNode) node).typeName().text())) {
+                    memberDeclarationNodes.add(node);
+                }
+            } else if (node.kind().equals(CONST_DECLARATION) && commonTypesList.contains(((ConstantDeclarationNode)
+                    node).variableName().text())) {
+                memberDeclarationNodes.add(node);
+            } else if (node.kind().equals(ENUM_DECLARATION) && commonTypesList.contains(((EnumDeclarationNode)
+                    node).identifier().text())) {
+                memberDeclarationNodes.add(node);
+            }
+        }
+
+        if (authConfig.isApiKeysConfig()) {
+            Token typeName = AbstractNodeFactory.createIdentifierToken("ApiKeysConfig");
+            NodeList<Node> nodeList = createNodeList(generateApiKeysConfigRecordFields(authConfig));
+            RecordTypeDescriptorNode recordTypeDescriptorNode =
+                    NodeFactory.createRecordTypeDescriptorNode(createToken(RECORD_KEYWORD),
+                            createToken(OPEN_BRACE_PIPE_TOKEN), nodeList, null,
+                            createToken(CLOSE_BRACE_PIPE_TOKEN));
+            TypeDefinitionNode typeDefinitionNode = createTypeDefinitionNode(metadataNode,
+                    createToken(PUBLIC_KEYWORD), createToken(TYPE_KEYWORD), typeName,
+                    recordTypeDescriptorNode, createToken(SEMICOLON_TOKEN));
+            memberDeclarationNodes.add(typeDefinitionNode);
+        }
+
+        ModulePartNode moduleNode = createModulePartNode(createEmptyNodeList(),
+                createNodeList(memberDeclarationNodes), createToken(EOF_TOKEN));
+        TextDocument textDocument = TextDocuments.from(EMPTY_STRING);
+        SyntaxTree configTypeSyntaxTree = SyntaxTree.from(textDocument);
+        return configTypeSyntaxTree.modifyWith(moduleNode);
+    }
+
+    private TypeDefinitionNode constructConnectionConfig(ModuleMemberDeclarationNode node, AuthConfig authConfig) {
```

Comment:
If possible please add documentation comments above every function for later readability and consistency. Please apply wherever possible

Reply:
Added in https://github.com/ballerina-platform/graphql-tools/commit/d6020522edacade9f7e6537cff02da89fa0b1b21